modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -30,7 +30,18 @@ func (k Keeper) InitInterchainAccount(ctx sdk.Context, connectionID, counterpart
 		return sdkerrors.Wrap(err, "unable to bind to newly generated portID")
 	}
 
-	msg := channeltypes.NewMsgChannelOpenInit(portID, icatypes.VersionPrefix, channeltypes.ORDERED, []string{connectionID}, icatypes.PortID, icatypes.ModuleName)
+	connectionEnd, err := k.channelKeeper.GetConnection(ctx, connectionID)
+	if err != nil {
+		return err
+	}
+
+	metadata := icatypes.NewMetadata(icatypes.Version, connectionID, connectionEnd.GetCounterparty().GetConnectionID(), "")
```

Comment:
Okay, this is the new JSON metadata in the version field.

I get it and like the design